src/segment_time.go

```diff
@@ -15,15 +15,36 @@ const (
 )
 
 func (t *tempus) enabled() bool {
+	// if no date set, use now(unit testing)
+	if t.CurrentDate.IsZero() {
+		t.CurrentDate = time.Now()
+	}
+	segmentTemplate := t.props.getString(SegmentTemplate, "")
+	if segmentTemplate != "" {
+		template := &textTemplate{
+			Template: segmentTemplate,
+			Context:  t,
+		}
+		t.templateText = template.render()
+		return len(t.templateText) > 0
+	}
 	return true
 }
 
 func (t *tempus) string() string {
-	timeFormatProperty := t.props.getString(TimeFormat, "15:04:05")
-	return time.Now().Format(timeFormatProperty)
+	return t.getFormattedText()
 }
 
 func (t *tempus) init(props *properties, env environmentInfo) {
 	t.props = props
 	t.env = env
 }
+
+func (t *tempus) getFormattedText() string {
```

Comment:
hi @lnu , I was just playing around with this and found a way that we could reuse the existing property if we're interested in that.  I'm just throwing the idea out there in case we want to go this direction. Fully understand if it's too magic and we'd rather go with a new property that is explicitly a template + dto.

```go
func (t *tempus) string() string {
	timeFormatProperty := t.props.getString(TimeFormat, "15:04:05")
	timeStr := time.Now().Format(timeFormatProperty)
	template := &textTemplate{
		Template: timeStr,
	}
	return template.render()
}
```

with that code either one of these is valid:
```json
// prints the old behavior:
"time_format": "January 02, 2006 (MST)"

// prints in lower case:
"time_format": "{{\"January 02, 2006 (MST)\" | lower}}"
```

Reply:
it makes more sense for a time segment not to have a custom property. While doing the change, I thought we could even have a "generic" template segment because you can do all in one template for the time: ` date | \"January 02, 2006 (MST)\" | lower`

Reply:
Indeed.